addons/sale_service/models/sale_order_line.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from itertools import groupby
+
+from odoo import api, fields, models
+from odoo.tools import format_amount
+from odoo.tools.sql import column_exists, create_column
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    # used to know if generate a task and/or a project, depending on the product settings
+    is_service = fields.Boolean("Is a Service", compute='_compute_is_service', store=True, compute_sudo=True)
```

Comment:
The question is: why is this field stored anyway?

Reply:
@KangOl To be honest, I don't really know it was the case since 13.0 (and perhaps even before). 😬 
The field has been stored by that [commit](https://github.com/odoo/odoo/commit/7c1a1d15eebc50420e376138f45064059242640e) in the following PR: #18228. 😬 

Maybe to improve a little bit the performance since the field is used in many domains in the Project app? 🤔 🤷🏽‍♂️

Reply:
Then, just add a `product_type` related field and use the domain `[('product_type', '=', 'service')]`. Or just add a search method on the field.

But this is for another task.

Reply:
Good idea, we will do that in another task. Thanks for your review 😁!